src/webworker.js

```diff
@@ -1,22 +1,24 @@
 importScripts('./pyodide.js')
 
 onmessage = async function(e) {
-  await loadPyodide({indexURL : '{{ PYODIDE_BASE_URL }}'});
-  const data = e.data;
-  for (let key of Object.keys(data)) {
-    if (key !== 'python') {
-      // Keys other than python must be arguments for the python script.
-      // Set them on self, so that `from js import key` works.
-      self[key] = data[key];
+  try {
+    const data = e.data;
+    for (let key of Object.keys(data)) {
+      if (key !== 'python') {
+        // Keys other than python must be arguments for the python script.
+        // Set them on self, so that `from js import key` works.
+        self[key] = data[key];
+      }
     }
-  }
 
-  try {
-    self.postMessage(
-        {results : await self.pyodide.runPythonAsync(data.python)});
+    if (typeof self.__pyodideLoading === "undefined") {
+      await loadPyodide({indexURL : '{{ PYODIDE_BASE_URL }}'});
+    }
+    let res = await self.pyodide.runPythonAsync(data.python);
+    self.postMessage({results : res});
   } catch (e) {
     // if you prefer messages with the error
-    self.postMessage({error : e.message});
+    self.postMessage({error : e.message + '\n' + e.stack});
```

Comment:
I'm not sure what's best to do here. `e.message` by itself is not enough as it's hard to debug without the stack trace.

Now for a good fraction of Python exceptions, the `e.stack` also appears to contain the message, meaning that it would be included twice with the current version.
However if I remove `e.message` I'm concerned that we then won't get the actual error message for JS exceptions.

So I guess repeating the message twice is better then potentially missing it. I'm not if there is a better way of managing it.

Reply:
> repeating the message twice is better then potentially missing it

I think repeating the message is fine and normal.